runtime/near-vm-logic/src/mocks/mock_external.rs

```diff
@@ -209,6 +248,46 @@ impl External for MockedExternal {
     fn validator_total_stake(&self) -> Result<Balance> {
         Ok(self.validators.values().sum())
     }
+
+    #[cfg(feature = "protocol_feature_function_call_weight")]
+    fn distribute_unused_gas(&mut self, gas: Gas) -> GasDistribution {
+        let gas_weight_sum: u128 =
+            self.gas_weights.iter().map(|(_, GasWeight(weight))| *weight as u128).sum();
+        if gas_weight_sum != 0 {
+            let gas_per_weight = (gas as u128 / gas_weight_sum) as u64;
+
+            let mut distribute_gas = |metadata: &FunctionCallActionIndex, assigned_gas: u64| {
+                let FunctionCallActionIndex { receipt_index, action_index } = metadata;
+                if let Some(Action::FunctionCall(FunctionCallAction { ref mut gas, .. })) = self
+                    .receipts
+                    .get_mut(*receipt_index)
+                    .and_then(|receipt| receipt.actions.get_mut(*action_index))
+                {
+                    *gas += assigned_gas;
+                } else {
+                    panic!("Invalid index for assigning unused gas weight");
```

Comment:
maybe also print what was the index (and also some other identifier - transaction/receipt id?)

( I assume that this should never ever happen, right?)

Reply:
yeah, this is currently unreachable, but just have a static error message to help debug if the functionality ever changes to re-order actions or receipts.

Using formatting machinery to include indices comes at a cost, can you confirm this is ideal before I make the change?

Reply:
but wait -  this cost would be paid only if the panic actually triggers, right?